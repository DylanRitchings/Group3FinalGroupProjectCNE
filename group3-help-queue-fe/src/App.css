.screenDiv {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  text-align: center;
}

.queue {
  width: 44%;
  height: 70%;
  position: absolute;
  top: 10%;
  left: 5%;
  border: 2px solid black;
}

.completed {
  width: 44%;
  height: 70%;
  position: absolute;
  top: 10%;
  right: 5%;
  border: 2px solid black;
}