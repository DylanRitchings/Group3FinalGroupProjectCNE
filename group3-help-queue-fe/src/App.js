import './App.css';
import CompletedTickets from './completedTickets';
import QueuedTickets from './queuedTickets';
import WelcomeUser from './welcomeUser';
import { useState } from 'react';

function App() {

  const [tickets, setTickets] = useState([
    {

        id:1,
        usersName: "Luke Foster",
        time: "17:14",
        ticketDescription: "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Error perferendis porro consequuntur illum omnis quas ex incidunt eius eos sed magni earum iusto maiores laboriosam numquam maxime, quidem nobis sint.",
        ticketTitle: "Docker Problem"

    },
    {
        id:2,
        usersName: "Jack Smith",
        time: "17:36",
        ticketDescription: "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Error perferendis porro consequuntur illum omnis quas ex incidunt eius eos sed magni earum iusto maiores laboriosam numquam maxime, quidem nobis sint.",
        ticketTitle: "Terraform Problem"
    },
    {
        id:3,
        usersName: "Ben Wellens",
        time: "11:25",
        ticketDescription: "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Error perferendis porro consequuntur illum omnis quas ex incidunt eius eos sed magni earum iusto maiores laboriosam numquam maxime, quidem nobis sint.",
        ticketTitle: "Login Problem"
    },
    {
        id:4,
        usersName: "Sam Matterson",
        time: "14:52",
        ticketDescription: "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Error perferendis porro consequuntur illum omnis quas ex incidunt eius eos sed magni earum iusto maiores laboriosam numquam maxime, quidem nobis sint.",
        ticketTitle: "500 Server error"
    }
  ])

  return (
    <div className="screenDiv">
      <WelcomeUser/>
    <div className="queue">
      <QueuedTickets tickets={tickets} setTickets={setTickets}/>
    </div>
    <div className="completed">
      <CompletedTickets/>
    </div>
    </div>
  );
}

export default App;
