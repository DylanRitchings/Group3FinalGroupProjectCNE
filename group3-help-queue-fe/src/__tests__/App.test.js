import {render, screen, cleanup, waitForElement} from '@testing-library/react';
import App from '../App';
import axiosMock from 'axios';
import Fetch from '../Fetch';
import axios from 'axios';
import { TestScheduler } from "@jest/core";

import '@testing-library/jest-dom';
import {expect, it } from '@jest/globals';
import {create} from 'react-test-renderer';

it(`Take a snapshot each time it is rendered`,()=>{
    const snapShot = create(<App/>).toJSON()
    console.log(snapShot);
    expect(snapShot).toMatchSnapshot();
})


afterEach(cleanup);

it('mocking axios request', async () => {

});