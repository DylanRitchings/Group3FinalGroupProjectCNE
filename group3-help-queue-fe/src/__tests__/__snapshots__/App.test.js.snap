// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should take a snapshot every time it's redered 1`] = `
<p>
  Please wait.... we are getting your request
</p>
`;

exports[`Take a snapshot each time it is rendered 1`] = `
<p>
  Please wait.... we are getting your request
</p>
`;
